mempool/src/tests/integration_tests.rs

```diff
@@ -150,6 +154,172 @@ async fn vfn_middle_man_test() {
         .await;
 }
 
+/// Tests when a node skips an ack
+#[tokio::test]
+async fn test_skip_ack_rebroadcast() {
+    let all_txns = test_transactions(0, 2);
+    let all_txns = all_txns.as_slice();
+    let mut node = MempoolTestFrameworkBuilder::single_validator();
+    let (other_peer_network_id, other_metadata) =
+        validator_mock_connection(ConnectionOrigin::Inbound, &ALL_PROTOCOLS);
+    // Add transactions
+    node.assert_txns_not_in_mempool(&all_txns[0..1]);
+    node.add_txns_via_client(&all_txns[0..1]).await;
+
+    node.connect_self(other_peer_network_id.network_id(), other_metadata.clone());
+
+    // After we drop, we should rebroadcast successfully
+    node.drop_next_network_msg(other_peer_network_id.network_id())
+        .await;
+    node.verify_broadcast_and_ack(other_peer_network_id, &all_txns[0..1])
+        .await;
+}
+
+/// Tests when a node gets disconnected.  Node should pick up after the second sending
+/// TODO: also add an outbound test to ensure it'll broadcast all transactions again
+#[tokio::test]
+async fn test_interrupt_in_sync_inbound() {
+    let all_txns = test_transactions(0, 2);
+    let all_txns = all_txns.as_slice();
+    for (mut node, (other_peer_network_id, other_metadata)) in inbound_node_combinations() {
+        // After we connect, partial should make progress
```

Comment:
nit: what do we mean by `partial`? The subset of transactions in `all_txns`?

Reply:
We use `partial` below too, which I'm still not sure about

Reply:
Ah, not the greatest comment.  The first txn will be sent.